### ISSUE TITLE
`add-tag/` and `remove-tag/` routes no longer work with private tags

### ISSUE THREAD

**Issue by ArthurHoaro** [shaarli/Shaarli#1608]
Private tags are prefixed with a dot (e.g.: `.category1`), except with the new routing system the web server interpret routes such as `https://domain.tld/add-tag/.category1` as accessing a dotfile, which is is forbidden by documented configuration.

I don't really want to change the web server directives, so I suggest that we change the route instead.
For example (if it works, didn't test it):

  * `/add-tag/.category1` -> `/tags/.category1/add`
  * `/remove-tag/.category1` -> `/tags/.category1/remove`

Or

  * `/add-tag/.category1` -> `/add-tag-.category1`
  * `/remove-tag/.category1` -> `/remove-tag-.category1`

**Comment by nodiscc** [comment]
I think the old syntax is more intuitive and more in line with API routes https://shaarli.github.io/api-documentation/#links-tag, so I don't know about this.

We could change the documeneted server config to just prevent access to "common" dotfiles and directories (version control, `.ssh`... this is what I do [server-wide](https://github.com/nodiscc/xsrv/blob/master/roles/apache/templates/etc_apache2_conf-available_security.conf.j2#L65)). Of course we can always miss some cases where users would have stashed a `.my-bank-credentials` file in their Shaarli documentroot, but I think it's unlikely.

I'd rather break/change server configuration than routes, the recommended config has already been changed recently, we could try to get it right once and for all before 1.0.0

What do you think?

**Comment by kcaran** [comment]
I think the fix for Apache is relatively simple. Using 'FilesMatch' instead of 'LocationsMatch' allows the routes to begin with periods but still prevents dotfiles from being accessed. I tested it out on my site.

```
39,42c39,41
<     <LocationMatch "/\.">
<         # Prevent accessing dotfiles
<         RedirectMatch 404 ".*"
<     </LocationMatch>
---
>     <FilesMatch "^\.(.+)$">
>         Require local
>     </FilesMatch>
```

```
$ curl -i https://shaarli.kcaran.com/add-tag/.net
HTTP/1.1 302 Found
Date: Fri, 23 Oct 2020 13:26:18 GMT
Server: Apache/2.4.6 (CentOS) OpenSSL/1.0.2k-fips mod_fcgid/2.3.9 PHP/7.4.9
X-Powered-By: PHP/7.4.9
Set-Cookie: shaarli=t2c6rnjk5n20tqvhvr19u7ntt9; domain=shaarli.kcaran.com
Expires: Thu, 19 Nov 1981 08:52:00 GMT
Cache-Control: no-store, no-cache, must-revalidate
Pragma: no-cache
Last-Modified: Fri, 23 Oct 2020 13:26:18 GMT
Cache-Control: post-check=0, pre-check=0
Location: /?searchtags=.net
Content-Length: 0
Content-Type: text/html; charset=UTF-8

$ curl -i https://shaarli.kcaran.com/.test
HTTP/1.1 403 Forbidden
Date: Fri, 23 Oct 2020 13:26:24 GMT
Server: Apache/2.4.6 (CentOS) OpenSSL/1.0.2k-fips mod_fcgid/2.3.9 PHP/7.4.9
Content-Length: 207
Content-Type: text/html; charset=iso-8859-1

<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>403 Forbidden</title>
</head><body>
<h1>Forbidden</h1>
<p>You don't have permission to access /.test
on this server.</p>
</body></html>
```

**Comment by ArthurHoaro** [comment]
I think you're right @nodiscc, and as you said we already have an ongoing (unreleased) change in our nginx default config.

Thanks for your input @kcaran, we can update the documentation/`.htaccess` with it.

**Comment by ArthurHoaro** [comment]
Fixed by #1630 and #1628
